app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -0,0 +1,80 @@
+module ManageIQ::Providers
+  module Hawkular
+    class MiddlewareManager::RefreshParser
+      def self.ems_inv_to_hashes(ems, options = nil)
+        new(ems, options).ems_inv_to_hashes
+      end
+
+      def initialize(ems, _options = nil)
+        @ems = ems
+        @eaps = []
+        @data = {}
+        @data_index = {}
+      end
+
+      def ems_inv_to_hashes
+        @data[:middleware_servers] = fetch_middleware_servers
+        fetch_deployments
+        @data
+      end
+
+      def fetch_middleware_servers
+        @data[:middleware_servers] = []
+        @ems.feeds.map do |feed|
+          @ems.eaps(feed).map do |eap|
+            @eaps << eap
+            server = parse_middleware_server(eap)
+            @data[:middleware_servers] << server
+            @data_index.store_path(:middleware_servers, :by_nativeid, server[:nativeid], server)
```

Comment:
There is a helper method in other providers (particularly the cloud ones) called `process_collection`, which sets up a nice collection pattern.  If you can copy that pattern, it will mirror the other providers.